src/ip-port-to-multiaddr.js

```diff
@@ -21,25 +21,29 @@ function ipPortToMultiaddr (ip, port) {
     throw errCode(new Error(`invalid ip provided: ${ip}`), errors.ERR_INVALID_IP_PARAMETER)
   }
 
-  port = parseInt(port)
+  if (typeof port === 'string') {
+    port = parseInt(port)
+  }
 
   if (isNaN(port)) {
     throw errCode(new Error(`invalid port provided: ${port}`), errors.ERR_INVALID_PORT_PARAMETER)
   }
 
-  if (new Address4(ip).isValid()) {
+  try {
+    // Test valid IPv4
+    new Address4(ip) // eslint-disable-line no-new
     return multiaddr(`/ip4/${ip}/tcp/${port}`)
-  }
+  } catch (_) {}
```

Comment:
If you're ignoring the error you can use optional catch binding here:

```suggestion
  } catch {}
```